src/worker.cc

```diff
@@ -154,26 +155,47 @@ void Worker::newUnixSocketConnection(evconnlistener *listener, evutil_socket_t f
 }
 
 Status Worker::listenTCP(const std::string &host, int port, int backlog) {
-  sockaddr_in sin{};
-  sin.sin_family = AF_INET;
-  evutil_inet_pton(AF_INET, host.data(), &(sin.sin_addr));
-  sin.sin_port = htons(port);
-  int fd = socket(AF_INET, SOCK_STREAM, 0);
-  int sock_opt = 1;
-  if (setsockopt(fd, SOL_SOCKET, SO_REUSEADDR, &sock_opt, sizeof(sock_opt)) < 0) {
-    return Status(Status::NotOK, evutil_socket_error_to_string(EVUTIL_SOCKET_ERROR()));
-  }
-  // to support multi-thread binding on macOS
-  if (setsockopt(fd, SOL_SOCKET, SO_REUSEPORT, &sock_opt, sizeof(sock_opt)) < 0) {
-    return Status(Status::NotOK, evutil_socket_error_to_string(EVUTIL_SOCKET_ERROR()));
+  char _port[6];
+  int af;
+  int rv, fd, sock_opt = 1;
+
+  if (strchr(host.data(), ':')) {
+    af = AF_INET6;
+  } else {
+    af = AF_INET;
+  }
+  snprintf(_port, sizeof(_port), "%d", port);
+  struct addrinfo hints, *srv_info, *p;
+  memset(&hints, 0, sizeof(hints));
+  hints.ai_family = af;
+  hints.ai_socktype = SOCK_STREAM;
+  hints.ai_flags = AI_PASSIVE;
+
+  if ((rv = getaddrinfo(host.data(), _port, &hints, &srv_info)) != 0) {
```

Comment:
forget to call `freeaddrinfo` to free `addrinfo`

Reply:
good catch, will fix later.